code/DDSCodeTester.cpp

```diff
@@ -5693,21 +5722,65 @@ int main(
         int argc,
         const char** argv)
 {
-    if (argc != 6)
-    {
-        printf("Bad number of parameters\n");
-        exit(-1);
-    }
-
     // Also show log warnings to spot potential mistakes
     Log::SetVerbosity(Log::Kind::Warning);
     // Report filename and line number for debugging
     Log::ReportFilenames(true);
 
     int exit_code = 0;
-    if (!dds_permissions_test(argv[1], argv[2], argv[3], argv[4], argv[5]))
+    if (argc == 6)
+    {
+        if (!dds_permissions_test(argv[1], argv[2], argv[3], argv[4], argv[5]))
+        {
+            std::cout << "Error parsing persimission xml file" << std::endl;
+            exit_code = -1;
+        }
+    }
+    else if (argc == 2)
     {
-        std::cout << "Error parsing persimission xml file" << std::endl;
+        if (strncmp(argv[1], "Static", 6) == 0)
+        {
+            std::string file = argv[1];
+            DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+            if (factory->check_xml_static_discovery(file) != ReturnCode_t::RETCODE_OK)
```

Comment:
Does this API load the XML file or only checks if it is valid? Is it required to call `loadXMLFile` after calling this method?

Reply:
It does not matter. The `loadXMLFile` method is called internally. The user only has to set the file path or the data in the corresponding QoS setting.